packages/react-app/src/components/pages/Bounties/Bounty/claim.tsx

```diff
@@ -0,0 +1,154 @@
+import { useUser } from '@app/hooks/useUser';
+import { BountyClaimCollection, BountyCollection, StatusHistoryItem } from '@app/models/Bounty';
+import axios from '@app/utils/AxiosUtils';
+import { Alert, AlertIcon, Button, Flex, Modal, ModalBody, ModalCloseButton, ModalContent, ModalFooter, ModalHeader, ModalOverlay, Text, Textarea, Tooltip, useColorMode, useDisclosure } from '@chakra-ui/react';
+import { useRouter } from 'next/router';
+import { useState } from 'react';
+import { AuthContext } from '@app/context/AuthContext';
+import { useContext } from 'react';
+import { claimedBy, newStatusHistory } from '@app/utils/formUtils';
+import AccessibleLink from '@app/components/parts/AccessibleLink';
+import { CustomerContext } from '@app/context/CustomerContext';
+import { baseUrl } from '@app/constants/discordInfo';
+
+const BountyClaim = ({ bounty }: { bounty: BountyCollection }): JSX.Element => {
+	const { isOpen, onOpen, onClose } = useDisclosure();
+	const router = useRouter();
+	const { colorMode } = useColorMode();
+	const { user } = useUser();
+	const [message, setMessage] = useState<string>();
+	const [claiming, setClaiming] = useState(false);
+	const [claimed, setClaimed] = useState(false);
+
+	const confirmBounty = async () => {
+		if (message && user) {
+			try {
+				setClaiming(true);
+				const res = await axios.patch<void, any, BountyClaimCollection>(
+					`api/bounties/${bounty._id}/claim?customerId=${bounty.customerId}`
+					, {
+						claimedBy: claimedBy(user),
+						submissionNotes: message,
+						status: 'In-Progress',
+						statusHistory: newStatusHistory(bounty.statusHistory as StatusHistoryItem[]),
+					}
+				);
+				if (res.status === 200)	{
+					setClaimed(true);
+					await new Promise(resolve => setTimeout(resolve, 2000)).then(() => {
+						router.push('/' + bounty._id);
+						setClaimed(false);
+					});
+				}
+			} finally {
+				setClaiming(false);
+			}
+		} else {
+			throw new Error('Missing Message');
+		}
+	};
+	
+	return (
+		<>
+	  		{
+				bounty.discordMessageId
+				// If we have discord message info for the bounty, the user is taken into discord to claim
+					? <ClaimDiscord discordMessageId={bounty.discordMessageId} />
+				// else, they must use the web form variant
```

Comment:
discordMessageId is a required field, the create form stores as `''` in the web ui which coalesces to `false`. 

It would probably be better to have `discordMessageId` be an optional field instead. Same result but much more declarative